docs/architecture/ADR-003-QGB-ValSet.md

```diff
@@ -0,0 +1,272 @@
+
+## Decision
+Add the `ValSet` and `ValSetConfirm` type of messages in order to track the state of the validator set.
+
+## Detailed Design
+Since the QGB is only a one way bridge and is not transferring assets, it doesn't require the portions of the gravity module
+that recreate state from the bridged chain. We only need to keep things relating to signing over the validator set (such as
+`MsgSetOrchestratorAddress` and `MsgValsetConfirm`) and relayer queries (such as `ValsetConfirm` and `GetDelegateKeyByOrchestrator`).
+
+It works by relying on a set of signers to attest to some event on Celestia: the Celestia validator set.
+
+The QGB contract keeps track of the Celestia validator set by updating its view of the validator set with `updateValidatorSet()`.
+More than 2/3 of the voting power of the current view of the validator set must sign off on new relayed events, submitted with
+[`submitDataRootTupleRoot()`](https://github.com/celestiaorg/quantum-gravity-bridge/blob/980b9c68abc34b8d2e4d20ca644b8aa3025a239e/ethereum/solidity/src/QuantumGravityBridge.sol#L328).
+Each event is a batch of `DataRootTuples`, with each tuple representing a single data root (i.e. block header).
+Relayed tuples are in the same order as Celestia block headers.
+For more details, check the data commitment ADR.
+
+Finally, if there are no validator set updates for the unbonding window, the bridge must halt.
+
+### When are validator sets created
+1. If there are no valSet requests, create a new one
+2. If there is at least one validator who started unbonding in current block. (we persist last unbonded block height in `hooks.go`)
+   This will make sure the unbonding validator has to provide an attestation to a new ValSet
+   that excludes him before he completely Unbonds. Otherwise, he will be slashed.
+3. If power change between validators of CurrentValSet and latest valSet request is > 5%
+
+### Message types
+We added the following messages types:
+
+#### Bridge Validator
+The `BridgeValidator` represents a validator's ETH address and its power.
+```protobuf
+message BridgeValidator {
+  uint64 power            = 1;
+  string ethereum_address = 2;
+}
+```
+It contains:
+- `power`: the voting power of the validator.
+- `ethereum_address`: the Ethereum address that will be used by the validator to sign messages.
+
+#### ValSet
+`Valset` is the Ethereum Bridge Multsig Set, each qgb validator also maintains an ETH key
+to sign messages, these are used to check signatures on ETH because of the significant gas savings.
+```protobuf
+message Valset {
+  uint64                   nonce   = 1;
+  repeated BridgeValidator members = 2 [(gogoproto.nullable) = false];
+  uint64                   height  = 3;
+}
+```
+It contains:
+- `nonce`: a unique number referencing the `ValSet`.
+- `BridgeValidator`: a list of [BridgeValidator](#Bridge-Validator) containing the current validator set.
+- `height`: the current chain height.
```

Comment:
These should honestly be comments in the protobuf, so that when it's copy-pasted the comments are there.

Reply:
Will put these in proto files also, then